api/services/user_activities_service.py

```diff
@@ -0,0 +1,103 @@
+from datetime import datetime
+from typing import Any, Optional
+
+from api import db, models
+from api.helpers import common_helper
+from api.utils.loggingutils import logger
+
+
+class UserActivitiesService:
+    def __init__(self, user, user_flow):
+        self.user_id = user.id
+        self.user_phone = user.phone
+        self.user_flow_id = user_flow.id
+        self.class_model = models.UserActivities
+
+    def handle_user_activities(self, json_data: dict[str, Any]):
+        try:
+            current_ist_time = common_helper.get_ist_timestamp()
+            contact_activities = json_data.get("contact", {}).get("fields", {})
+
+            for activity_key, _ in contact_activities.items():
+                if activity_key.strip() in contact_activities:
+                    user_activity = self.create_or_update_user_activity(
+                        activity_key, current_ist_time
+                    )
+                    if user_activity:
+                        db.session.add(user_activity)
+
+            db.session.commit()
+            logger.info(f"Captured user activity for {self.user_phone}.")
+        except Exception as e:
+            logger.error(
+                f"Error while capturing user activity for {self.user_phone}."
+                f"Error message: {e}"
+            )
+
+    def create_or_update_user_activity(
+        self, activity_key: str, current_ist_time: datetime
+    ) -> Optional[models.UserActivities]:
+        is_started = common_helper.check_activity_key(
+            activity_key, "activity_", "_started"
+        )
+        is_succeeded = common_helper.check_activity_key(
+            activity_key, "activity_", "_success"
+        )
+        is_completed = common_helper.check_activity_key(
+            activity_key, "activity_", "_completed"
+        )
+
+        if is_started:
+            return self.create_started_activity(activity_key, current_ist_time)
+        elif is_succeeded:
+            return self.update_succeeded_activity(current_ist_time)
+        elif is_completed:
+            return self.update_completed_activity(current_ist_time)
+        else:
+            return None
+
+    def create_started_activity(
```

Comment:
create Start activity should be similar to create activity, no? Can we do it like this in that case?
```suggestion
    def create_activity(
```

Reply:
Yes, Sure.